compiler/AST/primitive.cpp

```diff
@@ -799,6 +799,18 @@ initPrimitive() {
   prim_def(PRIM_SET_DYNAMIC_END_COUNT, "set dynamic end count", returnInfoVoid, true);
 
   prim_def(PRIM_GPU_KERNEL_LAUNCH, "gpu kernel launch", returnInfoVirtualMethodCall, true);
+  prim_def(PRIM_GPU_THREADIDX_X, "gpu threadIdx x", returnInfoInt32, true);
```

Comment:
I understand that these use nvidia terminology. Could you add comments here describing what they do without nvidia-jargon and also translating to opencl terms? The table at the top of https://github.com/chapel-lang/chapel/blob/main/doc/rst/developer/chips/17.rst might be of help here.

Reply:
Added in dc2d64ba9d8749d5567e7658d6fccb2864f4677b